geopandas/geoseries.py

```diff
@@ -170,7 +168,10 @@ def _wrapped_pandas_method(self, mtd, *args, **kwargs):
 
     def __getitem__(self, key):
         if isinstance(key, (slice, list, Series, np.ndarray)):
-            return super(GeoSeries, self).__getitem__(key)
+            block = self._data._block[key]
+            index = self.index[key]
+            return GeoSeries(SingleBlockManager(block, axis=index),
```

Comment:
This problem is still live in `align`, where we currently go through a densification process.  I'm not sure how best to handle this.